pylint/plugins/hass_constructor.py

```diff
@@ -0,0 +1,53 @@
+from astroid.node_classes import Const
+from astroid.nodes import ClassDef
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+class HassConstructorFormatChecker(BaseChecker):
+    """Check that __init__ return type is present."""
+
+    __implements__ = IAstroidChecker
+
+    name = "hass_constructor"
+    priority = -1
+    msgs = {
+        "W0006": (
+            '__init__ should have explicit return type "None"',
+            "hass-constructor-return",
+            "Used when __init__ has all arguments typed "
+            "but doesn't have return type declared",
+        ),
+    }
+    options = ()
+
+    def visit_functiondef(self, node):
+        """Called when a :class:`.astroid.node_classes.FunctionDef` node is visited.
+        See :mod:`astroid` for the description of available nodes.
+        :param node: The node to check.
+        :type node: astroid.node_classes.FunctionDef
```

Comment:
Why document it like this? Type annotations should be enough, aren't they? Same for the `register` function below.

Reply:
I just copied it from existing `hass_logger` plugin 🙂

I'll add proper type hints.